src/util/hass-translation.html

```diff
@@ -2,24 +2,13 @@
 
 <script>
 function getActiveTranslation() {
-  // Perform case-insenstive comparison since browser isn't required to
-  // report languages with specific cases.
-  const lookup = {};
-  /* eslint-disable no-undef */
-  Object.keys(window.translationMetadata).forEach((tr) => {
-    lookup[tr.toLowerCase()] = tr;
-  });
-
-  // Search for a matching translation from most specific to general
   function languageGetTranslation(language) {
-    const subtags = language.toLowerCase().split('-');
-
-    for (let i = subtags.length; i >= 1; i--) {
```

Comment:
Don't remove the matching of more general translations. You're breaking a preference of `en-US` matching `en`, etc.

Reply:
Not all browsers implement a complete navigator.languages. We need to support browsers that only provide one option.